src/contract.rs

```diff
@@ -137,9 +137,23 @@ pub fn execute(
                 return contract_error("only gp can redeem capital");
             }
 
-            let send = BankMsg::Send {
-                to_address: to.to_string(),
-                amount: coins(amount as u128, state.capital_denom),
+            let capital_marker =
+                ProvenanceQuerier::new(&deps.querier).get_marker_by_denom(&state.capital_denom)?;
+
+            let mut bank_send: Option<BankMsg> = None;
+            let mut marker_transfer: Option<CosmosMsg<ProvenanceMsg>> = None;
+            if capital_marker.marker_type == MarkerType::Coin {
+                bank_send = Some(BankMsg::Send {
+                    to_address: to.to_string(),
+                    amount: coins(amount as u128, &state.capital_denom),
+                });
+            } else {
+                marker_transfer = Some(transfer_marker_coins(
+                    amount as u128,
+                    &state.capital_denom,
+                    to,
+                    env.contract.address,
+                )?)
```

Comment:
Using the `?` operator here to "unwrap" the result from the `transfer_marker_coins` function which just has the actual message wrapped in a `StdResult`